services/venice-controller/src/main/java/com/linkedin/venice/pushmonitor/AbstractPushMonitor.java

```diff
@@ -500,7 +516,7 @@ public Pair<ExecutionStatus, String> getPushStatusAndDetails(String topic) {
           String errorMsg = "After waiting for " + elapsedTimeInSec + " seconds, resource assignment for: " + topic
               + " timed out, strategy=" + strategy.toString() + ", replicationFactor=" + replicationFactor + ", reason="
               + notReadyReason.get();
-          handleErrorPush(pushStatus, errorMsg);
```

Comment:
Refactoring idea:
It seems `handleErrorPush` and `handleCompletedPush` can be easily misused.
Maybe we should fold these two functions into `handleOfflinePushUpdate`.

Reply:
Hmm, the handleErrorPush and handleCompletedPush now is only used as the callback of the PushStatusCollector.
handleOfflinePushUpdate is the entry point to call to PushStatusCollector

Reply:
I see, then we can keep them.